

I have a debug patch which adds a BUG_ON to include/linux/list.h.  It causes
this:  

include/linux/list.h: In function `list_del':
include/linux/list.h:164: warning: implicit declaration of function `unlikely'

sparc64's bug.h should be including compiler.h.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-sparc64-akpm/include/asm-sparc64/bug.h |    2 ++
 1 files changed, 2 insertions(+)

diff -puN include/asm-sparc64/bug.h~sparc64-bug-needs-compiler-h include/asm-sparc64/bug.h
--- 25-sparc64/include/asm-sparc64/bug.h~sparc64-bug-needs-compiler-h	2004-06-19 21:51:24.696647624 -0700
+++ 25-sparc64-akpm/include/asm-sparc64/bug.h	2004-06-19 21:51:46.661308488 -0700
@@ -3,6 +3,8 @@
 #ifndef _SPARC64_BUG_H
 #define _SPARC64_BUG_H
 
+#include <linux/compiler.h>
+
 #ifdef CONFIG_DEBUG_BUGVERBOSE
 extern void do_BUG(const char *file, int line);
 #define BUG() do {					\
_
